extensions/arc/deployment/src/main/java/io/quarkus/arc/deployment/ObserverValidationProcessor.java

```diff
@@ -0,0 +1,63 @@
+package io.quarkus.arc.deployment;
+
+import java.util.Collection;
+
+import javax.inject.Inject;
+
+import org.jboss.jandex.AnnotationInstance;
+import org.jboss.jandex.AnnotationValue;
+import org.jboss.jandex.DotName;
+import org.jboss.jandex.IndexView;
+import org.jboss.logging.Logger;
+
+import io.quarkus.arc.processor.BeanDeploymentValidator;
+import io.quarkus.arc.processor.DotNames;
+import io.quarkus.arc.processor.ObserverInfo;
+import io.quarkus.deployment.annotations.BuildProducer;
+import io.quarkus.deployment.annotations.BuildStep;
+import io.quarkus.deployment.builditem.ApplicationArchivesBuildItem;
+
+/**
+ * Validates observer methods from application classes.
+ * If an observer listening for {@code @Initialized(ApplicationScoped.class)} is found, it logs a warning.
+ */
+public class ObserverValidationProcessor {
+
+    private static final Logger LOGGER = Logger.getLogger(ObserverValidationProcessor.class.getName());
+
+    @Inject
+    BuildProducer<BeanDeploymentValidatorBuildItem> validators;
+
+    @BuildStep
+    public void validateApplicationObserver(ApplicationArchivesBuildItem applicationArchivesBuildItem) {
+        // an index of all application classes
+        IndexView applicationClassesIndex = applicationArchivesBuildItem.getRootArchive().getIndex();
+
+        validators.produce(new BeanDeploymentValidatorBuildItem(new BeanDeploymentValidator() {
+
+            @Override
+            public void validate(ValidationContext context) {
+                Collection<ObserverInfo> allObservers = context.get(Key.OBSERVERS);
+                // do the validation for each observer that can be found within application classes
+                for (ObserverInfo observer : allObservers) {
+                    DotName declaringBeanDotName = observer.getDeclaringBean().getBeanClass();
+                    AnnotationInstance annotationInstance = observer.getObserverMethod().annotation(DotNames.INITIALIZED);
+                    if (applicationClassesIndex.getClassByName(declaringBeanDotName) != null &&
+                            annotationInstance != null &&
+                            annotationInstance.value().kind().equals(AnnotationValue.Kind.CLASS) &&
```

Comment:
This check is probably superfluous. If it's not `Kind.CLASS` then it's an error.

Reply:
> probably

That was my thinking exactly, I was just paranoid ;-)